src/context/tokenSelector.context.tsx

```diff
@@ -21,6 +21,7 @@ export const tokenSelectorContext = createContext({
     refetchXchainRoute: false as boolean,
     setRefetchXchainRoute: (value: boolean) => {},
     resetTokenContextProvider: () => {},
+    tokenPriceCompleted: false as boolean,
```

Comment:
nit: for clearness of understanding I'd prob rename this `isTokenPriceFetchingComplete`.

But it seems like an dangerous pattern to me

Reply:
I also was having doubts with this pattern, but I'm not sure of what refactor could I do to avoid the race condition between the two effects

Reply:
renamed

Reply:
renamed